cmds/core/init/init_linux.go

```diff
@@ -6,17 +6,57 @@ package main
 
 import (
 	"log"
+	"os"
+	"path/filepath"
 	"strconv"
+	"strings"
 	"syscall"
 
 	"github.com/u-root/u-root/pkg/cmdline"
+	"github.com/u-root/u-root/pkg/kmodule"
 )
 
+// installModules installs kernel modules (.ko files) from /lib/modules.
+// Useful for modules that need to be loaded for boot (ie a network
+// driver needed for netboot)
+func installModules() {
```

Comment:
I think this should be shared with whatever the modprobe command already does. At least it needs to be a library function in kmodule

cc @rminnich

Reply:
As far as the main difference between code here and modprobe, modprobe takes in options from its invocation cmdline whereas this takes in options from the kernel cmdline. Would you want modprobe to take in options from there as well? 

Otherwise, I believe you are right it would be better to use the kmodule modprobe function as it removes the call to os.Open here as well as installs module dependencies as needed.

Reply:
Share the code underneath, put the parsing up here? I haven't looked at the actual code, but I'm wary of the duplication.

Reply:
Having looked at the code, it seems there's minimal/no duplication. This code searches for module names to install (via glob), gets flags from /proc/cmdline via a function the cmdline package (which modprobe does not do), then uses kmodule package, similarly to modprobe, to install the module from the filename returned by glob.